games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe

```diff
@@ -0,0 +1,78 @@
+SUMMARY="A game inspired by the arcade classic Space Invaders"
+DESCRIPTION="SDLInvaders is a game inspired by the arcade classic Space Invaders using SDL"
+HOMEPAGE="https://sourceforge.net/projects/sdlinvaders/"
+COPYRIGHT="Niels Weber"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://master.dl.sourceforge.net/project/sdlinvaders/sdlinvaders/0.7.6/SDLInvaders-0.7.6.tar.bz2?viasf=1"
+CHECKSUM_SHA256="7646ce1d6f93e554f972f47529bcca01cb7228ccc6cd41fd5fa28d7e5e91fd75"
+SOURCE_FILENAME="SDLInvaders-0.7.6.tar.bz2"
+SOURCE_DIR="SDLInvaders"
+PATCHES="sdlinvaders-0.7.6.patchset"
+ADDITIONAL_FILES="sdlinvaders.rdef.in"
+
+ARCHITECTURES="all"
```

Comment:
Didn't check the recipe yet, but from your statement, clang doesn't work for 32bit, you are forcing a build with gcc2 here, that could be an issue, maybe marking x86_gcc2 as broken and add SECONDARY_ARCHITECTURES="x86" could solve this?

Reply:
> Didn't check the recipe yet, but from your statement, clang doesn't work for 32bit, you are forcing a build with gcc2 here, that could be an issue, maybe marking x86_gcc2 as broken and add SECONDARY_ARCHITECTURES="x86" could solve this?

It's actually the other way around, for some reason the game doesn't build with gcc11, which would create linking errors. At the same time though using clang in an x86 or x86_gcc2 enviroment will fail, but using gcc works. I have a feeling that it's actually a problem with gcc11, as it does compile with gcc8 and gcc2 without an error.

Reply:
>For some reason the game doesn't build with gcc11, which would create linking errors.
Could you please post the error messages?

Reply:
[sdlinvaders-0.7.6-build-gcc11.txt](https://github.com/haikuports/haikuports/files/7711765/sdlinvaders-0.7.6-build-gcc11.txt)
> > For some reason the game doesn't build with gcc11, which would create linking errors.
> 
> Could you please post the error messages?

(The text file has the whole output of make, it contains quite a few compile warnings)

I already tried to use the --std=c90 flag, It didn't change anything

I ran this make CC=gcc on a nightly build.

I installed a r1beta3 x86_64 image on a virtual machine which comes with gcc8 and then it builds without a problem. But as soon as I ran SoftwareUpdater (which installs gcc11) this error occured again. That's also the reason why I thought that the x86_gcc4 version was compiling fine, I still had the old gcc version, it now doesn't build with clang, which doesn't find stddef.h, nor gcc11.

Reply:
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here